src/OpenTelemetry/Batch.cs

```diff
@@ -205,7 +210,7 @@ public bool MoveNext()
             }
 
             /// <inheritdoc/>
-            public void Reset()
+            public readonly void Reset()
```

Comment:
Newbie question: what does this mean?

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/write-safe-efficient-code#declare-readonly-members-for-mutable-structs